sdv/single_table/base.py

```diff
@@ -75,3 +95,638 @@ def fit(self, data):
+        """Filter the sampled rows that match the conditions.
+
+        If condition columns are float values, consider a match anything that
+        is closer than the given ``float_rtol`` and then make the value exact.
+
+        Args:
+            sampled (pandas.DataFrame):
+                The sampled rows, reverse transformed.
+            conditions (dict):
+                The dictionary of conditioning values.
+            float_rtol (float):
+                Maximum tolerance when considering a float match.
+
+        Returns:
+            pandas.DataFrame:
+                Rows from the sampled data that match the conditions.
+        """
+        for column, value in conditions.items():
+            column_values = sampled[column]
+            if column_values.dtype.kind == 'f':
+                distance = value * float_rtol
+                sampled = sampled[np.abs(column_values - value) <= distance]
+                sampled[column] = value
+            else:
+                sampled = sampled[column_values == value]
+
+        return sampled
+
+    def _sample_rows(self, num_rows, conditions=None, transformed_conditions=None,
+                     float_rtol=0.1, previous_rows=None):
+        """Sample rows with the given conditions.
+
+        Input conditions is taken both in the raw input format, which will be used
+        for filtering during the reject-sampling loop, and already transformed
+        to the model format, which will be passed down to the model if it supports
+        conditional sampling natively.
+
+        If condition columns are float values, consider a match anything that
+        is closer than the given ``float_rtol`` and then make the value exact.
+
+        If the model does not have any data columns, the result of this call
+        is a dataframe of the requested length with no columns in it.
+
+        Args:
+            num_rows (int):
+                Number of rows to sample.
+            conditions (dict):
+                The dictionary of conditioning values in the original format.
+            transformed_conditions (dict):
+                The dictionary of conditioning values transformed to the model format.
+            float_rtol (float):
+                Maximum tolerance when considering a float match.
+            previous_rows (pandas.DataFrame):
+                Valid rows sampled in the previous iterations.
+
+        Returns:
+            tuple:
+                * pandas.DataFrame:
+                    Rows from the sampled data that match the conditions.
+                * int:
+                    Number of rows that are considered valid.
+        """
+        if self._data_processor._dtypes is not None:
+            if conditions is None:
+                sampled = self._sample(num_rows)
+            else:
+                try:
+                    sampled = self._sample(num_rows, transformed_conditions)
+                except TypeError:
```

Comment:
why did we change this? This is for the case when certain models don't support conditional sampling so they raise a `NotImplementedError`. For example, `TVAE` and `CTGAN`

Reply:
Good catch, I thought it was a `TypeError` because we basically didn't accept that extra argument on those models (as we call the same method below).